packages/ui/src/models/visualization/flows/pipe-visual-entity.ts

```diff
@@ -70,14 +94,20 @@ export class PipeVisualEntity implements BaseVisualCamelEntity {
   }
 
   toJSON() {
-    return this.spec;
+    return this.pipe.spec;
```

Comment:
This is a bit strange, here we're just returning the `spec` section to be serialized, rather than the entire `Pipe` definition.

Reply:
I see it now, is the `Pipe` resource who serializes the entire object here:

```typescript
  toJSON(): PipeType {
    return this.pipe;
  }
```